R/bed_makewindows.r

```diff
@@ -52,13 +52,16 @@ bed_makewindows <- function(x, genome, win_size = 0,
   if (win_size == 0 && num_win == 0)
     stop('specify either `win_size` or `num_win`', call. = FALSE)
   
-  res <- purrr::by_row(x, split_interval, genome, win_size,
-           step_size, num_win, 
-           reverse, .collate = 'rows',
-           .labels = FALSE)
-  
-  res <- select(res, -.row)
-    
+  x <- rowwise(x)
+  #pass rows to split_intervals, by default do() will convert to list
+  res <- do(x, 
+            split_interval(as_data_frame(.),
```

Comment:
I can get around adding `.` to the globals by using standard evaluation:

```r
res <- do_(x, 
            ~split_interval(as_data_frame(.), 
                           genome,
                           win_size, 
                           step_size, 
                           num_win, 
                           reverse))
```
The call to `as_data_frame()` is used because `do()` is passing a `list` to `split_interval()`.
I'll experiment with using another approach that doesn't rely on `do()` to avoid the call to `as_data_frame()`